src/managers/shellManager.ts

```
@@ -0,0 +1,227 @@
+    const dateFormated = new Date()
+      .toISOString()
+      .split('T')[0]
+      .replace(/-/g, '');
+    const timestamp = new Date().getTime();
+
+    const stdOutFile = path.join(
+      DEFAULT_PROCESS_LOG_DIR,
+      `${dateFormated}_${timestamp}_${prefix}_stdout.log`
+    );
+    const stdErrFile = path.join(
+      DEFAULT_PROCESS_LOG_DIR,
+      `${dateFormated}_${timestamp}_${prefix}_stderr.log`
+    );
+
+    const uniqueID = randomUUID();
+    const wrappedCommand = `export UNIQUE_PROCESS_ID=${uniqueID} && ${command}`;
+    // Execute the command with the stdio redirected to the files.
+    const childProcess = execa(wrappedCommand, {
+      shell: true,
+      stdio: [
+        'ignore',
+        fs.openSync(stdOutFile, 'w'),
+        fs.openSync(stdErrFile, 'w'),
+      ],
+      reject: false,
+    });
+
+    const shellProcess: ShellProcess = {
+      command,
+      status: 'running',
+      pid: -1,
+      output: '',
+      startTime: new Date(),
+      stdErrFile,
+      stdOutFile,
+    };
+
+    // Handle the process output.
+    childProcess.stdout?.on('data', (data) => {
+      shellProcess.output += data.toString();
+    });
+
+    childProcess.on('exit', (code) => {
+      shellProcess.status = code === 0 ? 'done' : 'failed';
+    });
+
+    childProcess.unref(); // Allows the parent process to exit independently
+
+    // Retrieve the process ID from the system since execa will only return the child process PID and not the command PID.
+    // Kind of a hack, but it works.
+    // TODO: Find an equivalent way to find this on Windows.
+    const res = await execa(
+      'ps',
+      ['-o', 'pid,command', '|', 'grep', uniqueID],
+      {
+        shell: true,
+        reject: false,
+        stdout: 'pipe',
+      }
+    );
+    const line = res.stdout
+      .split('\n')
+      .find((line) => line.includes(`export UNIQUE_PROCESS_ID=${uniqueID}`));
+    if (!line) {
+      throw new Error('Child process not found');
```

not sure we should throw an error, or simply the fact that the process has been terminated.

---

I think for now I would rather have the CLI send an error in some edge cases, than having a silent error and having to dig the code to find why some shellprocesses would be terminated unexpectedly